### PR TITLE
MODSOURMAN-111 Implement MARC Record Analyzer

### PR DESCRIPTION
**Pull request by Anatolii-Starkov**
-implemented marc record type analyser
-covered it with test

### CODE REVIEW

File: src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+public class MarcRecordAnalyzer implements RecordAnalyzer {
+
+  private static final String LEADER_KEY = "leader";
+  private static final int RECORD_TYPE_INDEX = 5;
+  private static final Set<Character> BIB_CODES = new HashSet<>(asList('a', 'c', 'd', 'e', 'f', 'g', 'i', 'j', 'k', 'm', 'o', 'p', 'r', 't'));
+  private static final Set<Character> HOLDING_CODES = new HashSet<>(asList('u', 'v', 'x', 'y'));
+  private static final Set<Character> AUTHORITY_CODES = new HashSet<>(Collections.singletonList('z'));
+
+  /**
+   * Processes a json object to determinate type of record BIB, HOLDING, AUTHORITY, NA.
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java]: *determine

File: src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  public <T> T process(JsonObject record) {
+    final char recordTypeCode = getRecordTypeCode(record);
+    return (T) getMarcRecordType(recordTypeCode);
+  }
+
+  private char getRecordTypeCode(JsonObject record) {
+    try {
+      return isMarcRecord(record) ? record.getString(LEADER_KEY).charAt(RECORD_TYPE_INDEX) : Character.MIN_VALUE;
+    } catch (Exception e) {
+      return Character.MIN_VALUE;
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java]: consider adding some logging

File: src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java (hunk 1 above)
Comment (Anatolii-Starkov) [reply src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java]: thanks

File: src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java (hunk 2 above)
Comment (Anatolii-Starkov) [reply src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java]: Added

File: src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  public <T> T process(JsonObject record) {
+    final char recordTypeCode = getRecordTypeCode(record);
+    return (T) getMarcRecordType(recordTypeCode);//NOSONAR
+  }
+
+  private char getRecordTypeCode(JsonObject record) {
+    try {
+      return isMarcRecord(record) ? record.getString(LEADER_KEY).charAt(RECORD_TYPE_INDEX) : Character.MIN_VALUE;
+    } catch (Exception e) {
+      LOGGER.error(format("I could not get a record type character form the leader: %s", record.getString(LEADER_KEY)), e);
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java]: "I could not ..." - looks like your Analyzer has a personality

Comment (KaterynaSenchenko) [reply src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java]: besides, could you use parameterized logging?

Comment (Anatolii-Starkov) [reply src/main/java/org/folio/dataimport/util/marc/MarcRecordAnalyzer.java]: Sure

**Pull request merged**